### PR TITLE
Add some unit test for utils, storage, rafs, api and builder.

### PR DESCRIPTION
**Pull request by ZouTaooo**
## Relevant Issue (if applicable)
_If there are Issues related to this PullRequest, please list it._

## Details
_Please describe the details of PullRequest._

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by imeoer** [comment]
Great work, thanks! @ZouTaooo 
cc @jiangliu @adamqqqplay Let's take a look together.

**Comment by adamqqqplay** [comment]
@ZouTaooo Thanks for your contribution! It looks really good.

File: storage/Cargo.toml [hunk 1]
```diff
@@ -35,6 +35,7 @@ url = { version = "2.1.1", optional = true }
 vm-memory = "0.10"
 fuse-backend-rs = "^0.10.3"
 gpt = { version = "3.0.0", optional = true }
+toml = "0.5"
```
Comment (imeoer) [inline storage/Cargo.toml:38]: Maybe move it to `[dev-dependencies]`.

Comment (ZouTaooo) [reply storage/Cargo.toml]: Thanks for your review. Now, I have fixed it with new commit.

**Review by ZouTaooo — COMMENTED** [review]
The Smoke-Test found a test-case error. And the commit fix it.
Two seconds is an edge-case, but there is a resolution problem in linux timer. So the data will be updated with different index sometimes. So the new test case with two changes, first using the normal argument rather than edge case, secondly considering the timer delta error.

**Review by jiangliu — APPROVED** [review]
